lib/ansible/modules/packaging/os/yum.py

```diff
@@ -1084,25 +1103,18 @@ def ensure(module, state, pkgs, conf_file, enablerepo, disablerepo,
             module.run_command(yum_basecmd + ['clean', 'expire-cache'])
 
         my = yum_base(conf_file, installroot)
-        try:
-            if disablerepo:
+        if disablerepo:
+            try:
                 my.repos.disableRepo(disablerepo)
-            current_repos = my.repos.repos.keys()
-            if enablerepo:
-                try:
-                    my.repos.enableRepo(enablerepo)
-                    new_repos = my.repos.repos.keys()
-                    for i in new_repos:
-                        if not i in current_repos:
-                            rid = my.repos.getRepo(i)
-                            a = rid.repoXML.repoid
```

Comment:
Note: If this is causing the pep8 test to complain, it would be a valid place to use the # nopep8 comment (along with the comment explaining why we have to do this).